server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java

```diff
@@ -246,6 +250,42 @@ public void appendMessageWithSpecialCase(String messageName, String mailbox) thr
         appendMessage(messageName, "eml/htmlWithLongAndComplicatedContent.eml");
     }
 
+    @Given("^the user has a message \"([^\"]*)\" in the \"([^\"]*)\" mailbox with flags \"([^\"]*)\"$")
+    public void appendMessageWithFlags(String messageName, String mailbox, List<String> keywords) throws Exception {
+        Flags flags = getFlagsFromListOfKeywords(keywords);
+        appendMessage(messageName, flags);
+    }
+
+    private Flags getFlagsFromListOfKeywords(List<String> keywords) {
```

Comment:
Can't we get a factory method from a List<String?>

Please also specify explicitly that you do not filter, and accept draft.